main/src/library/MutList.flix

```diff
@@ -165,7 +427,18 @@ namespace MutList {
     ///
     /// If the given index `i` exceeds the length of the mutable list, the element is inserted at the last position.
     ///
-    pub def insert(v: mut MutList[a], x: a, i: Int32): Unit = ???
+    pub def insert(x: a, i: Int32, v: mut MutList[a]): Unit & Impure =
+        let MutList(a, l) = v;
+        let capacity = capacity(v);
+        let len = deref l;
+        if (capacity == 0)
+            a := Array.new(x, 1)
+        else if (i < len)
+            a := Array.update(i, x, deref a)
+        else
+            reserve(capacity, v);
+            a := Array.update(len, x, deref a);
+        l := len + 1
```

Comment:
This looks approximately right. However- you should *double* the size of the array if there is no more space. I don't think that is what is happening currently?

Reply:
The thought was that since `reserve` reserves room for `n` additional elements, then passing the number of elements in the list as `n` should double the size of the underlying array?

However, I use the capacity of the underlying array here instead of the actual number of elements, which means the array does not grow correctly. I am also missing some test cases for this logic (which I will add shortly).